RocketChat/Rocket.Chat
# fix: prevent broadcasting of system messages when hide setting is enabled

## Proposed changes (including videos or screenshots)


This change ensures that system messages are not broadcasted when the setting to hide system messages is enabled. By checking the Hide_System_Messages configuration, the system will prevent unnecessary message broadcasts, improving message handling efficiency. This fix guarantees adherence to user-defined visibility settings for system messages.

## Issue(s)


[SUP-559](https://rocketchat.atlassian.net/browse/SUP-559)

## Steps to test or reproduce


1. Set the Hide_System_Messages setting to include specific message types.
2. Send a message of one of these types.
3. Verify that the message is not broadcasted.

## Further comments

This update will enhance the efficiency of message handling by preventing the broadcast of system messages when configured to do so. This helps in maintaining cleaner logs and better performance in scenarios where certain system messages are not needed.


[SUP-559]: https://rocketchat.atlassian.net/browse/SUP-559?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
